### DISCUSSION TITLE
The Astro Repl

### DISCUSSION THREAD

**Discussion by louiss0** [withastro/roadmap#358] [Proposal]
Are you guys ever going to work on that REPL you guys made? You guys have not changed it in a while. I want to know how things work under the hood. https://astro.build/play. I know you guys have lots of stuff to do but. I'd like to see how things work under the hood. The **REPL** still has just Astro Beta stuff in it

**Comment by jasikpark** [comment]
I think we currently consider https://astro.new aka stackblitz and codesandbox and gitpod as workable stand-ins for now - you can build an example or reproduction about as quickly & it has a lot more effort put into enabling those setups.

**Reply by louiss0** [reply]
I want to see the compiled output so that I understand how the framework works. **Every other framework has one**! Having a REPL that shows compiled output is not about how the app works **but how the framework works**. Giving people the power to use the REPL would give users the power to spot bugs and tell the framework authors how to fix them. In fact, it would allow them to create libraries in a better way. There are times when people don't understand the mechanics of the tool that they are using, **when they try to use an aspect of the framework could look at the compiled output and see how and why a feature works the way it does**.
